src/views/UserResourceAdd.vue

```diff
@@ -0,0 +1,63 @@
+<template>
+  <v-container class="my-12" fluid>
+    <resource-availability-card
+      :resourceAvailability="resourceAvailability"
+    ></resource-availability-card>
+
+    <user-resource-form @handle-submit="handleSubmit"></user-resource-form>
+  </v-container>
+</template>
+
+<script setup>
+import { ref, computed, inject, watch } from "vue";
+import { useRoute, useRouter } from "vue-router";
+import { useStore } from "vuex";
+import ResourceAvailabilityCard from "@/components/Card/ResourceAvailabilityCard.vue";
+const { resourceId, userId } = defineProps({
+  resourceId: String,
+  userId: String,
+});
+const snackbarProvider = inject("snackbarProvider");
+const store = useStore();
+const route = useRoute();
+const router = useRouter();
+const allUsers = computed(() => store.getters["users/allUsers"]);
+const resourceAvailability = ref({});
+resourceAvailability.value = await store.dispatch(
+  "resources/fetchAvailabilityResourceById",
+  resourceId
+);
+const resourceQuantity = ref({});
+resourceQuantity.value = await store.dispatch(
+  "resources/fetchQuantityByResourceId",
+  resourceId
+);
+
+const resourceDetails = computed(
+  () => store.getters["resources/getResourceForm"]
+);
+
+watch(route, (newValue) => {
+  store.dispatch("resources/setResourceForm", {});
+});
+
+const handleSubmit = async () => {
+  const { userOption, quantity } = resourceDetails.value;
+  const selectedUser = allUsers.value.find((user) => user.name == userOption);
```

Comment:
Well I guess that multiple users should not have the same name as it should be unique, but how should I get the id when we are mapping to show the names of the users?

Reply:
When saving it to the store (or passing it in the handleSubmit function after we update it), you would pass the id instead of the name. You are fetching all the users and mapping them to their names only in UserResourceForm:
`const userOptions = computed(() =>
  store.getters["users/allUsers"].map((user) => user.name)
);`
Otherwise, you are right that for the time being the names are unique but soon there will be firstName and lastName instead of name. Those wouldn't be unique and could be matching (there are a lot of Ivan Ivanov's in this world). 
You can leave it for now as it is and when we make the change with the names, we'd need to think about how to change this so that it will be feasible.